src/main/java/com/questhelper/helpers/quests/thepathofglouphrie/MonolithPuzzle.java

```diff
@@ -0,0 +1,369 @@
+		shapes.put(ItemID.ORANGE_SQUARE, new ItemRequirement("Orange square", ItemID.ORANGE_SQUARE));
+		shapes.put(ItemID.YELLOW_SQUARE, new ItemRequirement("Yellow square", ItemID.YELLOW_SQUARE));
+		shapes.put(ItemID.GREEN_SQUARE, new ItemRequirement("Green square", ItemID.GREEN_SQUARE));
+		shapes.put(ItemID.BLUE_SQUARE, new ItemRequirement("Blue square", ItemID.BLUE_SQUARE));
+		shapes.put(ItemID.INDIGO_SQUARE, new ItemRequirement("Indigo square", ItemID.INDIGO_SQUARE));
+		shapes.put(ItemID.VIOLET_SQUARE, new ItemRequirement("Violet square", ItemID.VIOLET_SQUARE));
+
+		shapes.put(ItemID.RED_PENTAGON, new ItemRequirement("Red pentagon", ItemID.RED_PENTAGON));
+		shapes.put(ItemID.ORANGE_PENTAGON, new ItemRequirement("Orange pentagon", ItemID.ORANGE_PENTAGON));
+		shapes.put(ItemID.YELLOW_PENTAGON, new ItemRequirement("Yellow pentagon", ItemID.YELLOW_PENTAGON));
+		shapes.put(ItemID.GREEN_PENTAGON, new ItemRequirement("Green pentagon", ItemID.GREEN_PENTAGON));
+		shapes.put(ItemID.BLUE_PENTAGON, new ItemRequirement("Blue pentagon", ItemID.BLUE_PENTAGON));
+		shapes.put(ItemID.INDIGO_PENTAGON, new ItemRequirement("Indigo pentagon", ItemID.INDIGO_PENTAGON));
+		shapes.put(ItemID.VIOLET_PENTAGON, new ItemRequirement("Violet pentagon", ItemID.VIOLET_PENTAGON));
+
+		inStartZone = new ZoneRequirement(new Zone(
+			regionPoint(34, 17),
+			regionPoint(39, 25)
+		));
+
+		pushSouthernMonolithUp = new NpcStep(getQuestHelper(), NpcID.BIG_MONOLITH, regionPoint(37, 21), "Push the monolith north once");
+		pushSouthernMonolithUp.setMaxRoamRange(3);
+
+		getFirstShapes = new ObjectStep(getQuestHelper(), ObjectID.CHEST_49617, regionPoint(39, 25),
+			"Open the chest for some shapes");
+		getFirstShapes.addSubSteps(pushSouthernMonolithUp);
+
+		pushSWMonolithNorth = new NpcStep(getQuestHelper(), NpcID.BIG_MONOLITH, regionPoint(35, 29), "Push the south-west monolith north once");
+		pushSWMonolithNorth.setMaxRoamRange(2);
+
+		pushNWMonolithEast = new NpcStep(getQuestHelper(), NpcID.BIG_MONOLITH, regionPoint(35, 33), "Push the north-west monolith east once");
+		pushNWMonolithEast.setMaxRoamRange(2);
+
+		getSecondShapes = new ObjectStep(getQuestHelper(), ObjectID.CHEST_49617, regionPoint(36, 38),
+			"Open the chest for some more shapes. If the chest doesn't give you any shapes, drop the shapes in your inventory first, then click the chest, then pick up the shapes from the ground.");
+		getSecondShapes.addSubSteps(pushSWMonolithNorth, pushNWMonolithEast);
+
+		picklockChestForFirstKey = new ObjectStep(getQuestHelper(), ObjectID.CHEST_49614, regionPoint(33, 37), "Picklock the chest for a key");
+		picklockChestForFirstKey.addAlternateObjects(ObjectID.CHEST_49615);
+		picklockChestForFirstKey.addSubSteps(pushSWMonolithNorth, pushNWMonolithEast);
+
+		pushSmallMonolithSouth = new NpcStep(getQuestHelper(), NpcID.SMALL_MONOLITH, regionPoint(39, 33), "Push the small monolith south once");
```

Comment:
*Should* be fixed by [`3099c9b` (#1289)](https://github.com/Zoinkwiz/quest-helper/pull/1289/commits/3099c9bb78b8a1e2417d91af98d244884ed4477d), however it's untested so this would be great if you tested again on your next run through the quest